### ISSUE TITLE
myQ API: Error: 500 Internal Server Error

### ISSUE THREAD

**Issue by fredthedorfman** [hjdhjd/homebridge-myq#226]
**Describe The Problem:**
Getting 500 errors but the plugin is still working since I can still open/close my garage.  Errors seem to have started about 13 hours ago but just noticed now.  Occurs every 1-2 minutes or so in the log.  I did try restarting homebridge, no change.

Edit: Tried opening my garage just now and It didn't respond through the Home app.  Waited a minute and tried again which worked. 

**To Reproduce:**

**Logs:**

```
[myQ] myQ API: Error: 500 Internal Server Error
[myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
```

**Homebridge Configuration:**

```json
{
            "email": "***************",
            "password": "*****************",
            "platform": "myQ"
}
```

**Screenshots:**

**Environment:**

* **Homebridge Version:** 1.1.7
* **Node Version:** 14.15.3
* **Homebridge-myQ Plugin Version**: 2.4.1
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** iPhone 12, iOS 14.3
* **Operating System and OS Version:** macOS 11.1

**Comment by magicman0978** [comment]
Having the same problem after update to version 2.4.1

**Comment by hjdhjd** [comment]
I'm not seeing any issues. Give it time and see if myQ is doing more server side activities...the myQ infrastructure isn't always the most reliable, so it's difficult to tell what's a real issue versus the infrastructure.

If others see this, would appreciate knowing. Unless you're seeing errors every few seconds to a minute or so, I wouldn't worry about it.

**Comment by DMBlakeley** [comment]
I am seeing the same issue with warnings quite frequently, 30 sec to 2 min. Have same MacOS environment as @fredthedorfman. Updated Node to 14.15.4 with no change. Garage door is responding so quite strange.

**Comment by hjdhjd** [comment]
@DMBlakeley If the garage door is responding, you're fine. It's an infrastructure thing. I'm now starting to see stuff like this:

```
0|homebridge  | [1/5/2021, 12:01:21 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:01:21 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:02:37 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:02:37 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:03:18 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:03:18 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:04:59 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:04:59 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:05:52 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:05:52 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:07:33 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:07:33 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:09:52 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:09:52 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:10:45 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:10:45 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:10:48 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:10:48 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:17:01 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:17:01 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:19:44 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:19:44 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:21:14 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:21:14 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:21:29 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:21:29 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:24:48 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:24:48 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:26:05 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:26:05 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:27:10 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:27:10 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:27:13 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:27:13 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:27:16 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:27:16 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
0|homebridge  | [1/5/2021, 12:27:56 PM] [myQ] myQ API: Error: 500 Internal Server Error
0|homebridge  | [1/5/2021, 12:27:56 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
```

This is not atypical, and *usually* is the result of infrastructure issues / changes on the myQ server end. It eventually resolves on it's own. It's only a **problem** if it's preventing you from using the garage door or getting status.

**Comment by phrz** [comment]
I know it's wishful thinking but it would be wonderful if someone discovered an undocumented local API for these things...

**Comment by hjdhjd** [comment]
@phrz It's called getting up and pressing the button yourself. :smile:

**Comment by magicman0978** [comment]
I have 3 homebridge running. The myQ plugin running version 2.4.0 does not
have this error 500 but one I updated to 2.4.1 started having this problem.
I don’t think it’s a myQ server issue but restore something to do with the
update of the plugin.

**Comment by hjdhjd** [comment]
@magicman0978 You'll have to prove that one to me. :smile: Reboot all three instances. Then see what happens.

**Comment by ultrathew** [comment]
Also seeing the same error. Sometimes I get it every couple of minutes, other times I get it a few times a minute. Otherwise, all seems to be working as expected. I've been able to open and close via the Home app a few times in the past hour.

**Comment by magicman0978** [comment]
Sorry, after the reboot the issue started showing up on the other server is
running the older version of the plug-in.

**Comment by furenberger** [comment]
I will try to get a MITM set up on android tonight to investigate the URLs, unless @hjdhjd you have any ideas? I don't have it running today since we shut off the other myq repo.

**Comment by rayr007** [comment]
Just noting that I see the same thing in my logs despite everything still seemingly working (although at this point it contributes nothing). 😄 I did update node from v12 to v14 yesterday so am glad to see it's not just me.
```
[1/5/2021, 2:15:38 PM] [myQ] myQ API: Error: 500 Internal Server Error
[1/5/2021, 2:15:38 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
[1/5/2021, 2:16:31 PM] [myQ] myQ API: Error: 500 Internal Server Error
[1/5/2021, 2:16:31 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
[1/5/2021, 2:17:12 PM] [myQ] myQ API: Error: 500 Internal Server Error
[1/5/2021, 2:17:12 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
[1/5/2021, 2:17:52 PM] [myQ] myQ API: Error: 500 Internal Server Error
[1/5/2021, 2:17:52 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
[1/5/2021, 2:18:45 PM] [myQ] myQ API: Error: 500 Internal Server Error
[1/5/2021, 2:18:45 PM] [myQ] myQ API: Unable to update device status from myQ servers. Acquiring a new security token and retrying later.
```

**Comment by phrz** [comment]
@furenberger I'm not super into this but if there were an undocumented local API, I'm inclined to agree with @hjdhjd that the app probably only goes through the cloud API, but MITM away to confirm that. If there is a local API, i.e. for debugging purposes, it could be discovered with some well-thought-out probing, but I don't think the app's network activity would give you any hints.

**Comment by furenberger** [comment]
I get the idea about local vs cloud API.  My theory is the CLOUD API URL changed for getting the auth token, which is why I saw that myq-cloud URL in my pi.hole logs.

**Comment by hjdhjd** [comment]
@furenberger That isn't why this error is happening. If it were, nothing would work, and things clearly do work. :smile: This isn't the first, nor will it be the last, time these errors come up: **they happen regularly**.

What you're seeing mostly relates to the new v6 API revision myQ has been experimenting with lately. You'll see references to myq-cloud.com, myq-prod.servicebus.windows.net, and others - they're slowly upgrading their infrastructure to something that scales but it likely won't be meaningfully complete for a few months, given they're trying to incorporate some of their acquisitions (cameras, locks) in the process. All that said - none of this impacts the fact that the v5 API, which we use, continues to work.

Given that things work for people - closing this issue. If there are real failures people are experiencing (not just log messages! :smile:), please open a new issue.

**Comment by MikeDProducer** [comment]
This is more of an issue the usual hiccups. I'm seeing lots of people (including myself) post about it across many different boards. Something is wrong at MyQ or they've changed their APIs.

**Comment by hjdhjd** [comment]
@MikeDProducer I see the same logs, but things are still working for me, and it seems the vast majority of folks. All of us who maintain the major myQ plugins across different platforms are looking into it...but for the time being, it’s more of a logging annoyance than a real issue.

We’ll see how the next few hours shake out.

**Comment by rayr007** [comment]
It’s more than just logs. Half the time commands going out to do actions like “open door” don’t make it...

**Comment by hjdhjd** [comment]
Thanks, I'll look into it.

**Comment by MikeDProducer** [comment]
> @MikeDProducer I see the same logs, but things are still working for me, and it seems the vast majority of folks. All of us who maintain the major myQ plugins across different platforms are looking into it...but for the time being, it’s more of a logging annoyance than a real issue.
> 
> 
> 
> We’ll see how the next few hours shake out.

Mine works like 80-90% of the time at the moment. I feel like my homebridge instance is running a tad slower as these lines keep popping up in the logs. 

Thank you for your great work! I love this plugin!

**Comment by hjdhjd** [comment]
Appreciate the support. I'm accelerating some work to reengineer the plugin for the newest API updates, just to get past this hump...but there's quite a few changes to work through, and some more reverse engineering that's still needed. They're clearly trying to make the API more stable and integrate their various acquisitions in a consistent way...it's just a royal pain for the rest of us. We'll get there. Just not the way I wanted to kick off 2021. :smile:

**Comment by markcsegal** [comment]
@hjdhjd Also seeing this quite often, let me know if you need logs.

**Comment by hjdhjd** [comment]
With tongue firmly in cheek let me say: the absolute last thing I need is more logs or “me toos”. :smile:

**Comment by hjdhjd** [comment]
Locking this issue guys - updates provided on the Homebridge Discord. Feel free to join the conversation.

Future readers: we’re aware there’s been yet another change in the myQ API and infrastructure. They seem to be transitioning to a new infrastructure and the errors we’re seeing are a result of the transition process and that they are deprecating the prior APIs we’ve used.  I’ve gotten some good logs of the new protocol...now it’s a function of implementing it over the coming days. Stay tuned.

**Comment by hjdhjd** [comment]
Fixed in v2.4.2.
